### ISSUE TITLE
Automatically add iptables rules when necessary for mbpr0

### ISSUE THREAD

**Issue by nielsreijers** [canonical/multipass#3468]
**What are you trying to do?**
I started a multipass VM on Fedora 39. The VM started correctly, but didn't have IPv4 internet access. IPv6 did work. The result was that apt worked (over IPv6), but snap didn't because api.snapcraft.io doesn't have an IPv6 address.

After some searching this turned out to be caused by the default iptables DROP policy for the FORWARD chain. Apparently many other distributions use default ACCEPT, but it's DROP on Fedora, so the solution to the issue was to add ACCEPT rules for the mpbr0 interface, after which everything worked as expected:

```
iptables -A FORWARD -i mpbr0 -j ACCEPT
iptables -A FORWARD -o mpbr0 -j ACCEPT
```

**What's your proposed solution?**
Since mbpr0 was created by multipass at some point, I think multipass should check at that point if these rules should be added, and either
- add them automatically,
- or print a big warning to inform the user he needs to add these rules if he wants the VM to have internet access.

I'm just getting started with multipass, so I can't really judge what the best place in the code is to make this change. I'm just using the default settings to quickly spin up an Ubuntu instance, but I'm aware there are some settings to connect different interfaces and there are probably scenarios where multiple mpbrX interfaces are used, so those would have to be considered as well.

**Additional context**
For some reason Fedora has ACCEPT for all chains in ip6tables, and only DROPs the FORWARD chain in iptables, which is why IPv6 worked.
This may make sense to people more familiar with iptables. Personally I'm not familiar enough with it to explain that difference, but if multipass would add these rules for IPv4 (when necessary), it may be a good idea to future proof it and do the same for IPv6 at the same time.

**Comment by luis4a0** [comment]
Hi @nielsreijers! Indeed, this needs to be addressed. We should fix this. In the meantime, we will document how to do this. Thanks!

**Comment by andrei-toterman** [comment]
Hey, @nielsreijers! Could you please elaborate on how you arrived at this issue? I tried a fresh install of Fedora 39 and I get `policy ACCEPT` for both `iptables -L FORWARD` and `ip6tables -L FORWARD`, so the default doesn't seem to differ from other distros. Could you please share the output of those commands? Could there be a chance that some time ago that was modified on your system?

**Comment by nielsreijers** [comment]
Hi @andrei-toterman,

Hmm this is interesting. I just did an `iptables -L FORWARD` and it was showing `(policy DROP)` on my new-ish Fedora 40 install.
Then I installed multipass again using snap, tried to start a vm with `multipass launch`, which got stuck in the starting stage. After I broke off the multipass command it now shows as status unknown in `multipass list`.

So I rebooted and now `iptables -L FORWARD` says `(policy ACCEPT)`... I have no idea why.
I don't have that much exotic stuff installed. Docker, K8s and minikube are the only ones I could imaging doing anything with the network configuration.

What I do remember is that I was playing with multipass again a few weeks ago on another fresh Fedora installation and had to add the ACCEPT rules again to get ip4 working, and that doing so did fix the issue a second time again.
Also, the first time I ran into this issue, I had no idea how iptables works, so I'm sure I didn't change the configuration myself.

Would you happen to know if there's any audit trail for iptables I could check to see why it has changed from DROP to ACCEPT just now?

**Comment by ricab** [comment]
Hi @nielsreijers, I believe docker is the culprit here, it sets the global forward policy to DROP. Do you happen to be using LXD as a backend for multipass? What does `multipass get local.driver` say?

**Comment by nielsreijers** [comment]
Hi @ricab , I did a fresh install, so it's back to qemu now, but yes, I was definitely using the LXD backend before.

Interesting that docker would do this. And apparently only for IPv4? I thought I read something about the default DROP policy being a Fedora thing, but I guess that's just a red herring then. Apologies.

Any idea as to why it's switched back to ACCEPT now? I did install multipass through snap yesterday, and can't remember if that was before or after I checked `iptables -L FORWARD`. But it was definitely set to DROP at some point earlier last night, and it's ACCEPT right now.
Is that something multipass would do during installation?

**Comment by ricab** [comment]
> Is that something multipass would do during installation?

No, I don't think so. I am not sure how that changed for you. Maybe you uninstalled docker? 

See this for more: https://documentation.ubuntu.com/lxd/en/latest/howto/network_bridge_firewalld/#prevent-connectivity-issues-with-lxd-and-docker

**Comment by tobe2098** [comment]
Not relevant anymore due to LXD deprecation.
